src/state/notifications/sagas.ts

```diff
@@ -97,35 +116,59 @@ export function* unsubscribeWalletSaga(action: UnsubscribeWalletAction) {
 
 export function* authenticateEmailSaga(action: AuthenticateEmailAction) {
   const { payload, meta } = action as AuthenticateEmailAction;
+
   try {
     const response = yield call(authenticateEmail, payload);
+
     if (response.result === Result.success) {
       yield put(authenticateEmailSuccess());
       if (meta?.onSuccess) {
-        meta.onSuccess();
+        meta.onSuccess(payload.session_token);
```

Comment:
or you can not reset session_token on authenticateEmailSuccess

Reply:
I would rather not parametrize authenticateEmailSaga only for that and I think it is good to clear used session token. I don't think that there is a pretty solution to this case

Reply:
I read the flow and I would go with not resetting the in redux AuthenticateEmailSuccessAction, this resetting is not needed anywhere, and this way we won't do some specific flow action an authenticate will remain the same. I think We shouldn't add `meta.onSuccess(payload.session_token)` because we have everywhere conception that meta.onSuccess has got same arguments on onSuccess action.  
With not resetting sessionToken we could also remove in src/screens/Notifications/UpdateEmailNotificationScreen.tsx 143 line             updateNotificationEmailSuccess(sessionToken); which I think is good thing as this action doesn't belong there as it is saga action for success

Reply:
`meta.onSuccess();` no need to pass session token

Reply:
And it is not passed, that's old commit

Reply:
i see it is `meta.onSuccess(payload.session_token);`

Reply:
yeah, here it was left. Done